libsolidity/ast/ASTAnnotations.h

```diff
@@ -233,10 +233,12 @@ struct TypeNameAnnotation: ASTAnnotation
 	TypePointer type = nullptr;
 };
 
-struct UserDefinedTypeNameAnnotation: TypeNameAnnotation
+struct IdentifierPathAnnotation: ASTAnnotation
 {
 	/// Referenced declaration, set during reference resolution stage.
 	Declaration const* referencedDeclaration = nullptr;
+	/// What kind of lookup needs to be done (static, virtual, super) find the declaration.
+	SetOnce<VirtualLookup> requiredLookup;
```

Comment:
Didn't we also want to add this to all other annotations that use `referencedDeclaration`?

Reply:
Yes, and we have done this already. It's already merged and all.

Reply:
ah, sorry!